inspire/modules/records/receivers.py

```diff
@@ -35,3 +32,33 @@ def populate_inspire_subjects(recid, json):
         if s.get('scheme', '') == 'INSPIRE' and s.get('term')
     ]
     json['facet_inspire_subjects'] = inspire_subjects
+
+
+@before_record_index.connect
+def match_valid_experiments(recid, json):
+    """Matches misspelled experiment names with valid experiments.
+       Tries to match with valid experiments by matching lowercased and
+       whitespace-free versions of known experiments.
+    """
+    # FIXME: This list is temporary. We should have a list that is generated
+    # from the current state of our data.
+    from .experiment_list import EXPERIMENTS_LIST
+    experiments_list_original = EXPERIMENTS_LIST
+    experiments_list = map(str.lower, experiments_list_original)
+    experiments_list = [exp.replace(' ', '') for exp in experiments_list]
+
+    experiments = json.get("accelerator_experiments")
+    if experiments:
```

Comment:
You can actually move the import here, so that there's no computation in case of no experiment.